### ISSUE TITLE
Categorical会采样到概率为0的取值的问题

### ISSUE THREAD

**Issue by yuxiaooye** [PaddlePaddle/Paddle#45555]
### bug描述 Describe the Bug

paddle版本：2.3.1 
IDE：BML CodeLab
```python
import paddle
from paddle.distribution import Categorical

paddle.seed(200)
logits = paddle.to_tensor([0.5, 0.5, 0.0])
dist = Categorical(logits)
a = dist.sample(shape=[1])
prob_a = dist.prob(a)
print(a)  # 注意：这里出现了bug，期望为【0】或【1】，但实际结果为概率为0的【2】
print(prob_a)  # 实际结果为0，若在生成概率分布时对logits使用softmax得到prob，期望结果不为0
```

为什么Categorical的sample()方法会采样到概率为0的取值【2】？
看到之前的issue #43971中提到paddle对logits使用softmax操作得到prob，如果真是这样，prob()方法的输出为0具有误导性。不过我查了一下`categorical.py`的源码，prob的计算仍是`self._prob = self.logits / dist_sum`，那么就不应该采样到【2】。烦请解惑！
